### PR TITLE
Cleanup and refactoring of Cipulot keyboards folder

### PR DESCRIPTION
**Pull request by Cipulot**
## Description

This commit includes cleanups and changes in includes of the `via_apc.c` file for affected keyboards, after the move of the file to a dedicated `common` folder in commit [#21233](https://github.com/qmk/qmk_firmware/pull/21233).

Here's a mode detailed list of the changes:
- cleanup of empty layers in the `keymap.c` files
- removal of features in `info.json` matching features in the default `info.json`
- delete of duplicated files in the `via` keymap folders of `via_apc.c` file and subsequent inclusion path update in `rules.mk`
- addition of `OPT` flags to increase scan rate through optimization
- fix of keycodes order in some keymaps
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cipulot/ec_23u/keymaps/numpad_6x4/keymap.c
[the remark this answers is not in the document]
Comment (Cipulot) [reply keyboards/cipulot/ec_23u/keymaps/numpad_6x4/keymap.c:1]: Good catch @lesshonor, that was an error while removing empty layers, will revert this one back to the use of `numpad_6x4` as before while deleting the actual empty layers.

[the remark this answers is not in the document]
Comment (Cipulot) [reply keyboards/cipulot/ec_23u/keymaps/numpad_6x4/keymap.c:1]: Fixed with commit.

**Comment by dunk2k** [comment]
Could this PR be targeted at the `develop` branch?
as per [PR Checklist](https://docs.qmk.fm/#/pr_checklist):
"keyboard updates and refactors (eg. to data driven) must go through `develop` to reduce `master` -> `develop` merge conflicts"

**Comment by Cipulot** [comment]
@dunk2k I didn't targeted this to `develop` simply because the base change was applied to `main` in the merge [21233](https://github.com/qmk/qmk_firmware/pull/21233).

It would create confusion to have multiple boards with the replicated code within the keymaps folder and the latest one being the only one with the latest changes requested to reduce code duplication.
Second opinions on the bast target for this more than welcome form other maintainers. Will act accordingly.

**Comment by Cipulot** [comment]
Retargeted to `develop`, since more cleanup was made, now it configures more with the requirements specified in the PR checklist.

**Comment by Cipulot** [comment]
Closing this since I've noticed that some of the EC code can be reworked and these changes are no longer needed, will do a separate and more comprehensive refactoring down the line (work already started on that).

**Pull request closed without merging**